jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -20,28 +20,31 @@
 import com.google.cloud.tools.jib.image.ImageReference;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.io.CharStreams;
+
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
 import java.util.function.Function;
 
-/** Calls out to the {@code docker} CLI. */
+/**
+ * Calls out to the {@code docker} CLI.
+ */
 public class DockerClient {
 
   private static final String DEFAULT_DOCKER_CLIENT = "docker";
+  private static final Map<String, String> DEFAULT_DOCKER_ENVIRONMENT = Collections.emptyMap();
```

Comment:
So this results in a change in behaviour as previously the docker command would have inherited the current environment.  This may actually break existing users who are using `minikube docker-env`.  I this should should default to `null` and the `defaultProcessBuilderFactory` should only set the environment if non-null.

Reply:
The default `DEFAULT_DOCKER_ENVIRONMENT` value was not there before. If no environment variable is passed, this is then used as a default one. Then, I simply add `dockerEnvironment` to the map of inherited ENV variables:
```
      ProcessBuilder pb = new ProcessBuilder(dockerCommand);

      Map<String, String> env = pb.environment();
      env.putAll(dockerEnvironment);
```

Therefore, I do not think this will introduce any breaking changes. 

You could argue, however, whether there is a point of passing an empty list as a default value. In this context, I kept consistent with the `dockerExecutable` configuration.

Ideally, I think, `DockerDaemonImage` should have no configuration defaults and call `newClient()` or `newClient(null, null)`. `DockerClient` would then check the parameters and would set defaults in case of `null`'s.

Reply:
> Therefore, I do not think this will introduce any breaking changes.

You're right — I misread. My apologies.

Reply:
No prob. Thanks for the feedback!